### ISSUE TITLE
System management is locked error

### ISSUE THREAD

**Issue by rexbinary** [openSUSE/openSUSE-repos#29]
This is my attempt at using the new repos.

```
❯ sudo zypper in openSUSE-repos-Tumbleweed
[sudo] password for rexbinary:
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following NEW package is going to be installed:
  openSUSE-repos-Tumbleweed

1 new package to install.
Overall download size: 13.0 KiB. Already cached: 0 B. After the operation, additional 1002.0 B will be used.
Continue? [y/n/v/...? shows all options] (y):
Retrieving: openSUSE-repos-Tumbleweed-20230725.c6c492e-1.1.x86_64 (Main Repository (OSS))            (1/1),  13.0 KiB
Retrieving: openSUSE-repos-Tumbleweed-20230725.c6c492e-1.1.x86_64.rpm ................................[done (13.0 KiB/s)]
Verifying .........................................................................................................[done]
Preparing .........................................................................................................[done]
(1/1) Installing: openSUSE-repos-Tumbleweed-20230725.c6c492e-1.1.x86_64 ...........................................[done]
Content of repo-debug.repo will be newly managed by zypp-services.
Storing old copy as {_sysconfdir}/zypp/repos.d/repo-debug.repo.rpmsave
Content of repo-source.repo will be newly managed by zypp-services.
Storing old copy as {_sysconfdir}/zypp/repos.d/repo-source.repo.rpmsave
Content of download.opensuse.org-oss.repo will be newly managed by zypp-services.
Storing old copy as {_sysconfdir}/zypp/repos.d/download.opensuse.org-oss.repo.rpmsave
Content of download.opensuse.org-tumbleweed.repo will be newly managed by zypp-services.
Storing old copy as {_sysconfdir}/zypp/repos.d/download.opensuse.org-tumbleweed.repo.rpmsave
System management is locked by the application with pid 9060 (/usr/libexec/zypp/zypp-rpm).
Close this application before trying again.
warning: %post(openSUSE-repos-Tumbleweed-20230725.c6c492e-1.1.x86_64) scriptlet failed, exit status 7
(1/1) Executing postin script for: openSUSE-repos-Tumbleweed-20230725.c6c492e-1.1.x86_64 .....................[attention]
```

This is the output from my attempt to restore:

```
❯ ls -la /etc/zypp/repos.d/*.rpmsave # review list of repos that will be restored
-rw-r--r-- 1 root root 175 Jul 18 22:51 /etc/zypp/repos.d/download.opensuse.org-oss.repo.rpmsave
-rw-r--r-- 1 root root 181 Jul 18 22:51 /etc/zypp/repos.d/download.opensuse.org-tumbleweed.repo.rpmsave
-rw-r--r-- 1 root root 158 Jul 18 22:51 /etc/zypp/repos.d/repo-debug.repo.rpmsave
-rw-r--r-- 1 root root 161 Jul 18 22:51 /etc/zypp/repos.d/repo-source.repo.rpmsave
❯ for file in /etc/zypp/repos.d/*.rpmsave; do echo mv $file `echo $file | sed -s "s/\.rpmsave//"`; done
mv /etc/zypp/repos.d/download.opensuse.org-oss.repo.rpmsave /etc/zypp/repos.d/download.opensuse.org-oss.repo
mv /etc/zypp/repos.d/download.opensuse.org-tumbleweed.repo.rpmsave /etc/zypp/repos.d/download.opensuse.org-tumbleweed.repo
mv /etc/zypp/repos.d/repo-debug.repo.rpmsave /etc/zypp/repos.d/repo-debug.repo
mv /etc/zypp/repos.d/repo-source.repo.rpmsave /etc/zypp/repos.d/repo-source.repo
❯ sudo zypper ref
Repository 'Main Repository (NON-OSS)' is up to date.
All repositories have been refreshed.
❯ zypper lr
Repository priorities are without effect. All enabled repositories share the same priority.

# | Alias                         | Name                      | Enabled | GPG Check | Refresh
--+-------------------------------+---------------------------+---------+-----------+--------
1 | download.opensuse.org-non-oss | Main Repository (NON-OSS) | Yes     | (r ) Yes  | Yes

```
I had to manually move the backed up repo files back into the correct directory.

```
❯ zypper lr
Repository priorities are without effect. All enabled repositories share the same priority.

# | Alias                            | Name                       | Enabled | GPG Check | Refresh
--+----------------------------------+----------------------------+---------+-----------+--------
1 | download.opensuse.org-non-oss    | Main Repository (NON-OSS)  | Yes     | (r ) Yes  | Yes
2 | download.opensuse.org-oss        | Main Repository (OSS)      | Yes     | (r ) Yes  | Yes
3 | download.opensuse.org-tumbleweed | Main Update Repository     | Yes     | (r ) Yes  | Yes
4 | repo-debug                       | openSUSE-Tumbleweed-Debug  | No      | ----      | ----
5 | repo-source                      | openSUSE-Tumbleweed-Source | No      | ----      | ----
```

**Comment by lkocman** [comment]
I'll talk to Michael Andres about some ideal solution. In general zypper refs shoud get it working even if call fails.
I suppose packagekit could have kicked-in.

**Comment by mlandres** [comment]
@rexbinary would you mind providing your /var/log/zypper.log

**Comment by lkocman** [comment]
@rexbinary I just confirmed with Michael that fix will land on the installer side. Thank you @mlandres for investigation.

**Comment by mlandres** [comment]
Fixed in libzypp-7.31.18
